File: android/app/src/main/kotlin/com/cryptokeno/app/data/models/Game.kt
package com.cryptokeno.app.data.models

import kotlinx.serialization.Serializable

@Serializable
data class GameState(
    val id: String,
    val gameType: String,
    val status: GameStatus,
    val players: List<Player>,
    val selectedNumbers: List<Int>,
    val winningNumbers: List<Int>,
    val prize: Double,
    val timestamp: Long
)

@Serializable
data class Player(
    val id: String,
    val username: String,
    val balance: Double,
    val bet: Double,
    val winnings: Double = 0.0
)

@Serializable
enum class GameStatus {
    WAITING, PLAYING, FINISHED, CANCELLED
}

@Serializable
data class LeaderboardEntry(
    val userId: String,
    val username: String,
    val rank: Int,
    val winnings: Double,
    val gamesPlayed: Int
)

File: ios/CryptoKeno/CryptoKeno/Models/Game.swift
import Foundation

struct GameState: Codable {
    let id: String
    let gameType: String
    let status: GameStatus
    let players: [Player]
    let selectedNumbers: [Int]
    let winningNumbers: [Int]
    let prize: Double
    let timestamp: Int64
}

struct Player: Codable {
    let id: String
    let username: String
    let balance: Double
    let bet: Double
    let winnings: Double
}

enum GameStatus: String, Codable {
    case waiting, playing, finished, cancelled
}

struct LeaderboardEntry: Codable {
    let userId: String
    let username: String
    let rank: Int
    let winnings: Double
    let gamesPlayed: Int
}
